mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistrationHandler.java

```diff
@@ -50,6 +51,11 @@
         this.groupRegistrations = new ConcurrentHashMap<>();
     }
 
+    GroupRegistration retrieveGroupRegistration(Group group) {
```

Comment:
why do you think the good action to perform is throwing here?

Reply:
Could be different reasons. After a refactoring maybe a group doesn't exist anymore and we try to redeliver an event to it, or we could have different versions with different groups running on different nodes. In those cases if the group doesn't exist we cannot redeliver the event and so that's why we throw an exception here in that case?

Reply:
Ok but GroupRegistrationHandler doesn't  know what (re)deliver is, right? So it's probably not a good thing to throw at that level. WDYT?